Freebsd: Redefinition of RSIZE_MAX

Hi, based on compilation logs of <https://gitlab.com/AOMediaCodec/SVT-AV1/-/jobs/8127838199>, freebsd defines RSIZE_MAX on their own.

```c
[5/270] Building C object CMakeFiles/safestringlib.dir/third_party/safestringlib/strncpy_s.c.o
FAILED: CMakeFiles/safestringlib.dir/third_party/safestringlib/strncpy_s.c.o 
/usr/lib/ccache/clang --target=x86_64-unknown-freebsd14.0 --sysroot=/opt/cross-freebsd -DARCH_X86_64=1 -DEN_AVX512_SUPPORT=1 -DEXCLUDE_HASH=0 -DHAVE_BUILTIN_EXPECT=1 -DHAVE_ELF_AUX_INFO=1 -DHAVE_VALGRIND_H=0 -DREPRODUCIBLE_BUILDS=1 -DSAFECLIB_STR_NULL_SLACK=1 -I/builds/AOMediaCodec/SVT-AV1/. -Wall -Wextra -Wformat -Wformat-security -Werror -Wshadow -pipe -fstack-protector-strong -mno-avx -g -std=gnu99 -flto=thin -fPIC -fvisibility=hidden -MD -MT CMakeFiles/safestringlib.dir/third_party/safestringlib/strncpy_s.c.o -MF CMakeFiles/safestringlib.dir/third_party/safestringlib/strncpy_s.c.o.d -o CMakeFiles/safestringlib.dir/third_party/safestringlib/strncpy_s.c.o -c /builds/AOMediaCodec/SVT-AV1/third_party/safestringlib/strncpy_s.c
In file included from /builds/AOMediaCodec/SVT-AV1/third_party/safestringlib/strncpy_s.c:7:
In file included from /builds/AOMediaCodec/SVT-AV1/third_party/safestringlib/safeclib_private.h:69:
/builds/AOMediaCodec/SVT-AV1/third_party/safestringlib/safe_lib.h:40:9: error: 'RSIZE_MAX' macro redefined [-Werror,-Wmacro-redefined]
#define RSIZE_MAX (256UL << 20) /* 256MB */
        ^
/opt/cross-freebsd/usr/include/stdint.h:72:9: note: previous definition is here
#define RSIZE_MAX (SIZE_MAX >> 1)
        ^
```

To reproduce:

```bash
$ docker run --rm -it registry.gitlab.com/aomediacodec/aom-testing/ubuntu2004-bsd
# git clone https://github.com/intel/safestringlib.git
# cmake -GNinja -S safestringlib/ -B build --toolchain /opt/cross-freebsd/toolchain.cmake -DCMAKE_BUILD_TYPE=Release
# ninja -C build
```

As a note, you can't compile the unit tests as there is the issue that freebsd already has a `memset_s()` with a different signature
```c
In file included from /safestringlib/unittests/test_memcmp16_s.c:13:
/safestringlib/include/safe_mem_lib.h:75:16: error: conflicting types for 'memset_s'
extern errno_t memset_s(void *dest, rsize_t dmax, uint8_t value);
               ^
/opt/cross-freebsd/usr/include/string.h:169:9: note: previous declaration is here
errno_t memset_s(void *, rsize_t, int, rsize_t);
        ^
```

--- a/include/safe_lib.h
+++ b/include/safe_lib.h
@@ -39,8 +39,10 @@ typedef size_t  rsize_t;
  */
 /* #define RSIZE_MAX (~(rsize_t)0)  - leave here for completeness */
 
+#ifndef RSIZE_MAX
 /* Bring back the standard */
 #define RSIZE_MAX        ( 256UL << 20 )     /* 256MB */
+#endif /* RSIZE_MAX */
 
 typedef void (*constraint_handler_t) (const char * /* msg */,
                                       void *       /* ptr */,

